nuxeo-features/nuxeo-search/nuxeo-search-core/src/main/java/org/nuxeo/ecm/platform/search/core/SavedSearchImpl.java

```diff
@@ -0,0 +1,199 @@
+public class SavedSearchImpl implements SavedSearch {
+
+    private DocumentModel doc;
+
+    public SavedSearchImpl(DocumentModel doc) {
+        this.doc = doc;
+    }
+
+    @Override
+    public String getId() {
+        return doc.getId();
+    }
+
+    @Override
+    public String getTitle() {
+        return getPropertyValue(SavedSearchConstants.TITLE_PROPERTY_NAME);
+    }
+
+    @Override
+    public Map<String, String> getNamedParams() {
+        List<Map<String, String>> paramsProperty = getPropertyValue(SavedSearchConstants.NAMED_PARAMS_PROPERTY_NAME);
+        Map<String, String> params = new HashMap<>();
+        for (Map<String, String> map : paramsProperty) {
+            params.put(map.get("key"), map.get("value"));
+        }
+        return params;
+    }
+
+    @Override
+    public String getQueryParams() {
+        return getPropertyValue(SavedSearchConstants.QUERY_PARAMS_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getQuery() {
+        return getPropertyValue(SavedSearchConstants.QUERY_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getQueryLanguage() {
+        return getPropertyValue(SavedSearchConstants.QUERY_LANGUAGE_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getPageProviderName() {
+        return getPropertyValue(SavedSearchConstants.PAGE_PROVIDER_NAME_PROPERTY_NAME);
+    }
+
+    @Override
+    public Long getPageSize() {
+        return getPropertyValue(SavedSearchConstants.PAGE_SIZE_PROPERTY_NAME);
+    }
+
+    @Override
+    public Long getCurrentPageIndex() {
+        return getPropertyValue(SavedSearchConstants.CURRENT_PAGE_INDEX_PROPERTY_NAME);
+    }
+
+    @Override
+    public Long getMaxResults() {
+        return getPropertyValue(SavedSearchConstants.MAX_RESULTS_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getSortBy() {
+        return getPropertyValue(SavedSearchConstants.SORT_BY_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getSortOrder() {
+        return getPropertyValue(SavedSearchConstants.SORT_ORDER_PROPERTY_NAME);
+    }
+
+    @Override
+    public String getContentViewData() {
+        return getPropertyValue(SavedSearchConstants.CONTENT_VIEW_DATA_PROPERTY_NAME);
+    }
+
+    @Override
+    public DocumentModel getDocument() {
+        return doc;
+    }
+
+    @Override
+    public void setTitle(String title) {
+        doc.setPropertyValue(SavedSearchConstants.TITLE_PROPERTY_NAME, title);
+    }
+
+    @Override
+    public void setNamedParams(Map<String, String> params) {
+        if (params != null) {
+            List<Map<String, String>> paramsProperty = getPropertyValue(SavedSearchConstants.NAMED_PARAMS_PROPERTY_NAME);
+            if (paramsProperty == null) {
+                paramsProperty = new ArrayList<>();
+            }
+
+            Map<String, String> variable;
+            for (String key : params.keySet()) {
+                String value = params.get(key);
+                variable = new HashMap<>(1);
```

Comment:
Won't work as `Collections.singletonMap` is immutable and two elements need to be added.

However `(1)` should be `(2)` (and I checked that "key" and "value" hash into different slots so it's optimal).